tabby-serial/src/components/serialTab.component.ts

```diff
@@ -82,12 +82,25 @@ export class SerialTabComponent extends BaseTerminalTabComponent<SerialProfile>
             this.session?.resize(this.size.columns, this.size.rows)
         })
         this.attachSessionHandler(this.session!.destroyed$, () => {
-            this.write(this.translate.instant(_('Press any key to reconnect')) + '\r\n')
-            this.input$.pipe(first()).subscribe(() => {
-                if (!this.session?.open) {
+            if (this.frontend) {
+                // Session was closed abruptly
+                if (this.config.store.terminal.behaviorOnSessionEnds === 'close') {
+                    // Close the tab
+                    this.destroy()
+                } else if (this.config.store.terminal.behaviorOnSessionEnds.startsWith('reconnect-or-')) {
```

Comment:
Where is the distinction between `reconnect-or-close` or `reconnect-or-keep-open` is handled? In this case, both fall through into `reconnect()`

Reply:
Well, for these two choices, i was thinking it as : reconnect ssh/serial/telnet and close or keep open the other kind of session.